### ISSUE TITLE
Google Play Store requires Android SDK 28 as of Nov 1st 2019

### ISSUE THREAD

**Issue by scottstensland** [meteor/meteor#10734]
Can meteor run using Android 28 ?   We currently use meteor 1.8.1 which is Android 27

below is an email I just received from Google Play Store saying they will require Android 28 for app updates after Nov 1st 2019

___________________________________________________________________________

[Action Required] Update your Android targetSdkVersion by November 1
Google Play no-reply-googleplay-developer@google.com
5:31 PM (1 hour ago)
Google Play
Developer update
Hello Google Play Developer,

This is a reminder that starting November 1, 2019, updates to apps and games on Google Play will be required to target Android 9 (API level 28) or higher. After this date, the Play Console will prevent you from submitting new APKs with a targetSdkVersion less than 28.

Configuring your app to target a recent API level ensures that users benefit from significant security and performance improvements, while still allowing your app to run on older Android versions (down to the minSdkVersion).

Action required

Please ensure that your apps are configured to target at least Android 9 (API level 28) by November 1, 2019. For technical advice on how to change your app's target API level to meet these requirements, refer to the migration guide.

Affected apps

The apps included below have one or more APKs—in production or testing tracks—that aren't currently targeting API level 28 or higher. Apps are listed with the maximum version code and corresponding targetSdkVersion. If you have more than 20 apps that could be affected in your account, please check the Play Console for a full list.

**Comment by rj-david** [comment]
mobile-config.js
```
App.setPreference('android-minSdkVersion', '22');
App.setPreference('android-targetSdkVersion', '28');
```

**Comment by wallind** [comment]
@scottstensland

Just wanted to weigh in here. I work for a small software development company 'Kyros LLC' (check us out 😏 ) and we just ran into this issue today trying to deploy an update for one of our mobile apps.

We are running Meteor version 1.8.0.2 and upon trying to push my apk bundle to the Play Store Google informed me I must change the targetSdkVersion to 28. I did and it broke things, to say the least.

TLDR: I was able to fix the issue by adding the following line to the AndroidManifest.xml as a child node of the <application> tag.
`android:usesCleartextTraffic="true"`
Here is the stack overflow article I used if it helps 🙃 https://stackoverflow.com/questions/45940861/android-8-cleartext-http-traffic-not-permitted

**Comment by mjmasn** [comment]
The above is terrible advice. Ensuring TLS is set up correctly on your server is the solution, not bypassing the clear text traffic restrictions that are there to protect user privacy and security. With the likes of Let's Encrypt, CloudFlare and co offering totally free certificates there's really no excuse in 2019 to still be using plain http:// connections.

**Comment by wallind** [comment]
My company uses Let's Encrypt for all of our servers. This was simply a quick fix to be able to continue developing. Additionally I'd recommend suggesting a specific solution instead of blasting others with general advice. Not every project requires secure data transfer anyway. If no identifying information is being collected than encrypted traffic isn't really necessary. This is especially true for some casual side projects.

**Comment by mjmasn** [comment]
https://doesmysiteneedhttps.com

As for specific solutions, as a starting point just Google "let's encrypt" or "tls" or "https" plus the name of your web host / server software for example "let's encrypt haproxy" and you'll find a guide or two. Then check your setup with https://www.ssllabs.com/ssltest/ It's not particularly difficult to score an A or A+ with 30-60 minutes of effort.

**Comment by Twisterking** [comment]
At the moment afaik you need `android:usesCleartextTraffic="true"` to be able to properly develop a cordova app locally - since all your assets are loaded from `http://localhost` without this setting you will end up at a blank screen! ;)

**Comment by mjmasn** [comment]
@Twisterking yep, but you can have a separate debug/AndroidManifest.xml for local development that'll get merged into the main/AndroidManifest.xml when you do a debug build. 

Not sure of the folder structure for Meteor/Cordova apps but you can see how it's done for the React Native template app here: https://github.com/facebook/react-native/blob/master/template/android/app/src/debug/AndroidManifest.xml

**Comment by sebakerckhof** [comment]
I verified that with the changes from: https://github.com/meteor/meteor/pull/10810
I can now run on Android 28 without needing to change the manifest. This will be part of Meteor 1.8.4+

**Comment by filipenevola** [comment]
We are going to have the first beta of Meteor 1.10 in the next days, including #10810 😉
